galsim/wfirst/wfirst_wcs.py

```diff
@@ -0,0 +1,571 @@
+        xtp1 = cos_theta1*np.sin(phi1.rad())*galsim.radians/sin_theta1
+        ytp1 = -cos_theta1*np.cos(phi1.rad())*galsim.radians/sin_theta1
+        xtp2 = cos_theta2*np.sin(phi2.rad())*galsim.radians/sin_theta2
+        ytp2 = -cos_theta2*np.cos(phi2.rad())*galsim.radians/sin_theta2
+        dxtp = xtp2 - xtp1
+        dytp = ytp2 - ytp1
+
+        # Finally have the ingredients for computing the position angle of this SCA Y axis.
+        pa_sca = np.arctan2(-cos_sca_rot*dxtp.rad()+sin_sca_rot*dytp.rad(),
+                             sin_sca_rot*dxtp.rad()+cos_sca_rot*dytp.rad())*galsim.radians
+
+        # Compute CD coefficients: extract the linear terms from the a_sip, b_sip arrays.  These
+        # linear terms are stored in the SIP arrays for convenience, but are defined differently.
+        # The other terms have been divided by the linear terms, so that these become pure
+        # multiplicative factors. There is no need to change signs of the SIP coefficents associated
+        # with odd powers of X! Change sign of a10, b10 because the tangent-plane X pixel coordinate
+        # has sign opposite to the detector pixel X coordinate, and this transformation maps pixels
+        # to tangent plane.
+        a10 = -a_sip[i_sca,1,0]
+        a11 = a_sip[i_sca,0,1]
+        b10 = -b_sip[i_sca,1,0]
+        b11 = b_sip[i_sca,0,1]
+
+        # Rotate by pa_fpa.
+        cos_pa = np.cos(pa_sca.rad())
+        sin_pa = np.sin(pa_sca.rad())
+        header['CD1_1'] = (cos_pa * a10 + sin_pa * b10, "partial of first axis coordinate w.r.t. x")
+        header['CD1_2'] = (cos_pa * a11 + sin_pa * b11, "partial of first axis coordinate w.r.t. y")
+        header['CD2_1'] = (-sin_pa * a10 + cos_pa * b10, "partial of second axis coordinate w.r.t. x")
+        header['CD2_2'] = (-sin_pa * a11 + cos_pa * b11, "partial of second axis coordinate w.r.t. y")
+        header['ORIENTAT'] = (pa_sca / galsim.degrees, 
+                              "position angle of image y axis (deg. e of n)")
+        header['LONPOLE'] = (phi_p / galsim.degrees,
+                             "Native longitude of celestial pole")
+
+        for i in range(n_sip):
+            for j in range(n_sip):
+                if i+j >= 2 and i+j < n_sip:
+                    sipstr = "A_%d_%d"%(i,j)
+                    header[sipstr] = a_sip[i_sca,i,j]
+                    sipstr = "B_%d_%d"%(i,j)
+                    header[sipstr] = b_sip[i_sca,i,j]
+
+        if not as_header:
+            wcs = galsim.FitsWCS(header=header)
+            wcs_list.append(wcs)
+        else:
+            wcs_list.append(header)
+
+    return wcs_list
+
+def findSCA(wcs_list, ra=None, dec=None, pos=None, include_border=False):
+    """
+    This is a subroutine to take a list of WCS (one per SCA) from galsim.wfirst.getWCS() and query
+    which SCA a particular real-world coordinate would be located on.  If the position is not
+    located on any of the SCAs, the result will be None.
```

Comment:
I'm confused - since the WCS does include the distortion model, I thought it would be included automatically when using findSCA()? (at least with ```include_border=False```; I agree that with it being ```True```, we are making a slightly incorrect extrapolation)  Have I missed something basic?